### DISCUSSION TITLE
How to improve grasping in a very simple scenario?

### DISCUSSION THREAD

**Discussion by HoangGiang93** [google-deepmind/mujoco#2309] [Asking for Help] [answered]
### Intro

Hi!

I am a researcher at Institute for Artificial Intelligence, University of Bremen. I use MuJoCo for my research on Physics Reasoning.

### My setup

MuJoCo 3.2.6
Ubuntu 20.04

### My question

Hello everyone, I've encountered a recurring issue that I usually resolve with temporary fixes, but I'd like to find a more permanent solution. The challenge is enhancing the grasping ability of a robot.

Specifically, I'm using the Panda robot hand to try and open a lid by gripping its knob, as well as picking up other loosely placed objects. As demonstrated in the video, the current friction is insufficient for the hand to maintain a secure grip on the knob, despite adjustments to the gripper force and attempts to increase the friction. The pick-and-place tasks are inconsistent, sometimes successful and sometimes not.

https://github.com/user-attachments/assets/6e7e5acb-de6a-4c17-a6a5-1282436c8279

Could anyone advise on how to properly manage this scenario?

### Minimal model and/or code that explain my question

[panda_with_task_board.zip](https://github.com/user-attachments/files/18232647/panda_with_task_board.zip)

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment, marked as answer]
Did you read this https://mujoco.readthedocs.io/en/latest/modeling.html#preventing-slip ?

**Reply by HoangGiang93** [reply]
Thank you for your quick response. Indeed, I review the documents thoroughly each time in an effort to avoid slipping. My issue, however, is that whenever I encounter a new scenario, I need to find a specific configuration tailored to that case. This feels quite like a workaround and leads to inconsistencies. Here’s an example of my current challenge: 

I’ve applied all the strategies from the documentation, they only work with the knob when I either set `noslip_iterations="1"` or higher, or use `cone="elliptic" impratio="10"`. Setting `noslip_iterations="1"` or higher seems to degrade the physics, while `cone="elliptic" impratio="10"` appears to work well at the moment, though I’m unsure if it might cause other issues later on.

https://github.com/user-attachments/assets/fcd8ac59-0c08-4b6f-8c0a-1b49fd2c68ac

So my question is, is it a good practice to automatically apply all of these strategies every time I need to simulate grasping and hope for getting lucky?

Merry Christmas to You :)

**Comment by yuvaltassa** [comment]
`cone="elliptic"` is always better, but a bit slower.
